src/NUnitFramework/framework/Constraints/Operators/BinaryOperator.cs

```diff
@@ -48,7 +48,7 @@ public override int LeftPrecedence
         {
             get
             {
-                return RightContext is CollectionOperator
+                return RightContext is CollectionOperator || RightContext is ExactCountOperator
```

Comment:
Could this be any uglier?!? It wouldn't be needed if ExactCountOperator were a CollectionOperator, of course. However, the original test is almost as bad. We increase the precedence of binary operators `And` and `Or` when followed by a collection operator. This is what allows us to write

```C#
    Exactly(3).EqualTo("Charlie").And.Exactly(1).EqualTo("Fred")
```
which means __almost but not quite__ the same thing as

```C#
    Exactly(3).EqualTo("Charlie").Or.EqualTo("Fred")
```

It would be better if we could figure out a way to avoid this context sensitivity in the grammar.